pkg/worker/verifier/producer_worker.go

```diff
@@ -164,12 +187,28 @@ func (pw *ProducerWorker) produceInner(n int64) (int64, []BadOffset, error) {
 		kgo.RequiredAcks(kgo.AllISRAcks()),
 		kgo.RecordPartitioner(kgo.ManualPartitioner()),
 	}...)
+
+	if pw.transactionsEnabled {
+		randId := uuid.New()
+		tid := "p" + randId.String()
+		log.Debugf("Configuring transactions with TransactionalID %s", tid)
+
+		opts = append(opts, []kgo.Opt{
+			kgo.TransactionalID(tid),
+			kgo.TransactionTimeout(2 * time.Minute),
+		}...)
+	}
+
 	client, err := kgo.NewClient(opts...)
 	if err != nil {
 		log.Errorf("Error creating Kafka client: %v", err)
 		return 0, nil, err
 	}
 
+	if pw.transactionsEnabled {
+		pw.transactionSTM = worker.NewTransactionSTM(context.Background(), client, pw.transactionSTMConfig)
```

Comment:
really like this approach of putting the transaction-isms into their own state class.